lib/Controller/MessageApiController.php

```diff
@@ -0,0 +1,165 @@
+	public function send(
+		int $accountId,
+		string $fromEmail,
+		string $subject,
+		string $body,
+		bool $isHtml,
+		array $to,
+		array $cc = [],
+		array $bcc = [],
+		array $references = [],
+	): DataResponse {
+		try {
+			$mailAccount = $this->accountService->find($this->userId, $accountId);
+		} catch (ClientException $e) {
+			$this->logger->error("Mail account #$accountId not found", ['exception' => $e]);
+			return new DataResponse($e, Http::STATUS_NOT_FOUND);
+		}
+
+		if($fromEmail !== $mailAccount->getEmail()) {
+			try {
+				$alias = $this->aliasesService->findByAliasEmailAndUserId($fromEmail, $this->userId);
+			} catch (DoesNotExistException $e) {
+				$this->logger->error("Alias $fromEmail for mail account $accountId not found", ['exception' => $e]);
+				// Cannot send from this email as it is not configured as an alias
+				return new DataResponse($e, Http::STATUS_NOT_FOUND);
+			}
+		}
+
+		$attachments = $this->request->getUploadedFile('attachments');
+		$numberOfAttachments = count($attachments['name'] ?? []);
+
+		$localAttachments = [];
+		$messageAttachments = [];
+		$attachmentErrors = false;
+		while ($numberOfAttachments > 0) {
+			$numberOfAttachments--;
+			$filedata = [
+				'name' => $attachments['name'][$numberOfAttachments],
+				'type' => $attachments['type'][$numberOfAttachments],
+				'size' => $attachments['size'][$numberOfAttachments],
+				'tmp_name' => $attachments['tmp_name'][$numberOfAttachments],
+			];
+			$file = new UploadedFile($filedata);
+			try {
+				$localAttachment = $this->attachmentService->addFile($this->userId, $file);
+				$messageAttachments[] = $localAttachment;
+				$localAttachments[] = ['type' => 'local', 'id' => $localAttachment->getId()];
+			} catch (UploadException $e) {
+				$this->logger->error('Could not convert attachment to local attachment', ['excpetion' => $e]);
+				$attachmentErrors = true;
+			}
+		}
+
+		if ($attachmentErrors) {
+			foreach ($localAttachments as $localAttachment) {
+				// Handle possible dangling local attachments
+				$this->attachmentService->deleteAttachment($this->userId, $localAttachment['id']);
+			}
+			return new DataResponse('Could not handle attachments', Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+
+		$message = new LocalMessage();
+		$message->setType(LocalMessage::TYPE_OUTGOING);
+		$message->setAccountId($accountId);
+		$message->setSubject($subject);
+		$message->setBody($body);
+		$message->setEditorBody($body);
+		$message->setHtml($isHtml);
+		$message->setSendAt($this->time->getTime());
+		$message->setType(LocalMessage::TYPE_OUTGOING);
+
+		if(isset($alias)) {
+			$message->setAliasId($alias->getId());
+		}
+		if(!empty($references)) {
+			$message->setInReplyToMessageId($references[0]); // needs DB changes I guess
+		}
+		if(!empty($attachments)) {
+			$message->setAttachments($messageAttachments);
+		}
+
+		$localMessage = $this->outboxService->saveMessage($mailAccount, $message, $to, $cc, $bcc, $localAttachments);
```

Comment:
The To, CC and BCC should be tested for valid length and format.

Reply:
not in this PR - that would mean changing the underlying code for the outbox too.